lib/gpu_memory_service/client/torch/module.py

```
@@ -1,162 +1,670 @@
 
-# =============================================================================
-# Public API - Registration and Materialization
-# =============================================================================
+    intervals: dict[str, list[tuple[int, int]]] = {}
+    for discovered_storage, allocation_id, start in located:
+        intervals.setdefault(allocation_id, []).append(
+            (start, start + discovered_storage.storage.nbytes())
+        )
+    for allocation_id, allocation_intervals in intervals.items():
+        ordered = sorted(allocation_intervals)
+        for previous, current in zip(ordered, ordered[1:], strict=False):
```

Ruff (0.15.21)

[warning] Prefer `itertools.pairwise()` over `zip()` when iterating over successive pairs

Replace `zip()` with `itertools.pairwise()`

(RUF007)